juju/client/connection.py

```diff
@@ -279,6 +281,7 @@ async def connect(
         self.addr = None
         self.ws = None
         self.endpoint = None
+        self.endpoints = None
```

Comment:
I'm wondering if we can get rid of `self.endpoint` altogether and then say `self.endpoints = [endpoint]`. That way we reduce the number of `if isinstance(endpoint, str)` etc.

Reply:
I can do this: self.endpoints = [endpoint]

But removing the self.endpoint is more complicated, because that variable is used to by the class to know to which endpoint in particular it is connected to

Reply:
mmmmm but if I do [endpoint], and endpoint is a list, then self.endpoints will be [["endpoint1", ...]]

Reply:
@davigar15 then we need to update `self.endpoint` to always be the one it's connected to. That way we're always in sync.

Reply:
@SimonRichardson That is updated in line 629, in the _connect function.

Every time libjuju reconnects, it executes the function _connect, and then this is executed:
         self.endpoint = result[2]

Reply:
Excellent.